tests/adapters/unified_tests/ip_management_test.py

```diff
@@ -22,92 +26,92 @@ class IpManagementTest(ConfiguredTestCase):
 
-        assert_that(vlan2345["ips"], has_length(3))
-        assert_that(vlan2345["ips"][0], equal_to({"address": "1.1.1.2", "mask": 24}))
-        assert_that(vlan2345["ips"][1], equal_to({"address": "1.1.1.4", "mask": 24}))
-        assert_that(vlan2345["ips"][2], equal_to({"address": "2.2.2.2", "mask": 24}))
+        assert_that(vlan2345.ips, has_length(3))
+        assert_that(str(vlan2345.ips[0]), equal_to("1.1.1.2/24"))
+        assert_that(str(vlan2345.ips[1]), equal_to("1.1.1.4/24"))
+        assert_that(str(vlan2345.ips[2]), equal_to("2.2.2.2/24"))
 
-        self.delete("/switches/{switch}/vlans/2345")
+        self.client.remove_vlan(number=2345)
 
     @skip_on_switches("juniper", "juniper_qfx_copper", "dell", "dell_telnet", "dell10g", "dell10g_telnet")
     def test_adding_unavailable_ips_and_various_errors(self):
-        self.post("/switches/{switch}/vlans", data={"number": 2345})
+        self.client.add_vlan(number=2345)
 
-        self.post("/switches/{switch}/vlans/2345/ips", data={"address": "1.1.1.1", "mask": "24"})
+        self.client.add_ip_to_vlan(vlan_number=2345, ip_network=IPNetwork("1.1.1.1/24"))
 
-        response = self.post("/switches/{switch}/vlans/3333/ips", data={"address": "1.1.1.1", "mask": "24"}, fail_on_bad_code=False)
-        assert_that(response.status_code, equal_to(404)) #vlan not found
+        with self.assertRaises(UnknownVlan):
+            self.client.add_ip_to_vlan(vlan_number=3333, ip_network=IPNetwork("1.1.1.1/24"))
 
-        response = self.post("/switches/{switch}/vlans/2345/ips", data={"address": "1.1.1.1", "mask": "24"}, fail_on_bad_code=False)
-        assert_that(response.status_code, equal_to(409))
+        with self.assertRaises(Conflict): #todo multiple exceptions: IPNotAvailable, IPAlreadySet
+            self.client.add_ip_to_vlan(vlan_number=2345, ip_network=IPNetwork("1.1.1.1/24"))
 
-        self.post("/switches/{switch}/vlans", data={"number": 2999})
-        response = self.post("/switches/{switch}/vlans/2999/ips", data={"address": "1.1.1.1", "mask": "24"}, fail_on_bad_code=False)
-        assert_that(response.status_code, equal_to(409))
+        self.client.add_vlan(number=2999)
+        with self.assertRaises(IPNotAvailable):
+            self.client.add_ip_to_vlan(vlan_number=2999, ip_network=IPNetwork("1.1.1.1/24"))
 
-        response = self.post("/switches/{switch}/vlans/2999/ips", data={"address": "1.1.1.2", "mask": "24"}, fail_on_bad_code=False)
-        assert_that(response.status_code, equal_to(409))
-        self.delete("/switches/{switch}/vlans/2999")
+        with self.assertRaises(IPNotAvailable):
+            self.client.add_ip_to_vlan(vlan_number=2999, ip_network=IPNetwork("1.1.1.2/24"))
 
-        response = self.delete("/switches/{switch}/vlans/1111/ips/1.1.1.1/24", fail_on_bad_code=False)
-        assert_that(response.status_code, equal_to(404)) #vlan not found
+        self.client.remove_vlan(number=2999)
```

Comment:
diff is out of sync because "self.client.remove_vlan(number=2999)" should not be there, will remove.